src/app/features/container/container.tsx

```diff
@@ -1,42 +1,174 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { Toaster } from 'react-hot-toast';
-import { Outlet, useLocation } from 'react-router-dom';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
 
+import { ChainID } from '@stacks/transactions';
+import { OnboardingSelectors } from '@tests/selectors/onboarding.selectors';
+import { Box } from 'leather-styles/jsx';
+import { token } from 'leather-styles/tokens';
+
+import { RouteUrls } from '@shared/route-urls';
 import { closeWindow } from '@shared/utils';
 
 import { useAnalytics, useInitalizeAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { LoadingSpinner } from '@app/components/loading-spinner';
+import { CurrentAccountAvatar } from '@app/features/current-account/current-account-avatar';
+import { CurrentAccountName } from '@app/features/current-account/current-account-name';
+import { SwitchAccountDialog } from '@app/features/dialogs/switch-account-dialog/switch-account-dialog';
 import { useOnSignOut } from '@app/routes/hooks/use-on-sign-out';
 import { useOnWalletLock } from '@app/routes/hooks/use-on-wallet-lock';
 import { useHasStateRehydrated } from '@app/store';
-import { useRouteHeaderState } from '@app/store/ui/ui.hooks';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+import { ContainerLayout } from '@app/ui/components/containers/container.layout';
+import { NetworkModeBadge } from '@app/ui/components/containers/headers/components/network-mode-badge';
+import { Header } from '@app/ui/components/containers/headers/header';
+import { Flag } from '@app/ui/components/flag/flag';
+import { Logo } from '@app/ui/components/logo';
+import { HamburgerIcon } from '@app/ui/icons/';
 
 import { useRestoreFormState } from '../popup-send-form-restoration/use-restore-form-state';
-import { SettingsDropdown } from '../settings-dropdown/settings-dropdown';
-import { SwitchAccountDrawer } from '../switch-account-drawer/switch-account-drawer';
-import { ContainerLayout } from './container.layout';
+import { Settings } from '../settings/settings';
+import { getDisplayAddresssBalanceOf, isKnownPopup, showAccountInfo } from './get-popup-header';
+import { getTitleFromUrl } from './get-title-from-url';
+import { TotalBalance } from './total-balance';
 
 export function Container() {
-  const [routeHeader] = useRouteHeaderState();
+  const [isShowingSwitchAccount, setIsShowingSwitchAccount] = useState(false);
+  const navigate = useNavigate();
   const { pathname } = useLocation();
   const analytics = useAnalytics();
   const hasStateRehydrated = useHasStateRehydrated();
 
+  const { chain, name: chainName } = useCurrentNetworkState();
+
   useOnWalletLock(() => closeWindow());
   useOnSignOut(() => closeWindow());
   useRestoreFormState();
   useInitalizeAnalytics();
 
   useEffect(() => void analytics.page('view', `${pathname}`), [analytics, pathname]);
 
+  function isHomePage() {
```

Comment:
This file is fairly large now and a bit hard to follow, better to pull some of this out into a custom hook?

Reply:
I'll indeed refactor this and make it more understandable and lightweight. The goal is get everything working and looking well, then make one last final pass and improve the code quality